File: sidecar/src/routes/diagnosticsRoutes.ts
import { Router } from 'express';
import { DiagnosticsService } from '../application/diagnosticsService';

export const createDiagnosticsRouter = (diagnostics: DiagnosticsService) => {
  const router = Router();

  router.get('/diagnostics/kms', async (_request, response) => {
    try {
      const result = await diagnostics.runKmsRoundTrip();
      response.status(result.ok ? 200 : 502).json(result);
    } catch (error) {
      response.status(500).json({
        ok: false,
        error: error instanceof Error ? error.message : String(error),
      });
    }
  });

  router.get('/diagnostics/cost-forecast', async (_request, response) => {
    try {
      const result = await diagnostics.runCostForecast();
      response.status(result.ok ? 200 : 500).json(result);
    } catch (error) {
      response.status(500).json({
        ok: false,
        error: error instanceof Error ? error.message : String(error),
      });
    }
  });

  return router;
};

File: mcp/floci_backend/api/diagnostics_routes.py
from fastapi import APIRouter
from fastapi.responses import JSONResponse
from typing import Dict, Any

from floci_backend.application.diagnostics_service import DiagnosticsService

def create_diagnostics_router(diagnostics_service: DiagnosticsService) -> APIRouter:
    router = APIRouter()

    @router.get('/diagnostics/kms')
    async def run_kms_diagnostic():
        result = await diagnostics_service.run_kms_round_trip()
        status_code = 200 if result.get('ok') else 502
        return JSONResponse(status_code=status_code, content=result)

    @router.get('/diagnostics/cost-forecast')
    async def run_cost_forecast():
        result = await diagnostics_service.run_cost_forecast()
        status_code = 200 if result.get('ok') else 500
        return JSONResponse(status_code=status_code, content=result)

    return router
